training/finetune_acft_hf_nlp/src/finetune/finetune.py

```diff
@@ -435,6 +444,44 @@ def finetune(args: Namespace):
     hf_task_runner = get_task_runner(task_name=args.task_name)()
     hf_task_runner.run_finetune(args)
 
+    process_name = os.environ.get('AZUREML_PROCESS_NAME', 'main')
+    logger.info(f"Process - {process_name}")
+    if process_name in {'main', 'rank_0'}:
+        # hotfix mlflow model deployment issue
+        mlflow_model_path = args.mlflow_model_folder
+        mlflow_conda_yaml = str(Path(mlflow_model_path, "conda.yaml"))
+        mlflow_req_file = str(Path(mlflow_model_path, "requirements.txt"))
+        try:
+            import yaml
+            logger.info("Updating mlflow model requirements")
+            with open(mlflow_conda_yaml, "r") as rptr:
+                mlflow_data = yaml.safe_load(rptr)
+            logger.info(mlflow_data)
+            for ele in mlflow_data["dependencies"]:
+                if type(ele) == dict and "pip" in ele:
+                    for idx in range(len(ele["pip"])):
+                        if ele["pip"][idx].startswith("mlflow==") or ele["pip"][idx].startswith("mlflow<"):
```

Comment:
you can filter the packages with mlflow and add the mlflow dependencies towards the end may be?
`if 'mlflow' in ele['pip']['idx']:
   continue
`
I am suggesting this because the current logic fails if `mlflow>...` is specified

Reply:
I will include `mlflow>` as well in logic. I can't filter everything with 'mlflow' in it as it will include other packages like azureml-evaluate-mlflow and currently we are doing this in a strict manner